model/action.go

```diff
@@ -0,0 +1,1070 @@
+		ctx, err := r.newPdfActionThreadFromDict(d)
+		if err != nil {
+			return nil, err
+		}
+		ctx.PdfAction = action
+		action.context = ctx
+		return action, nil
+	case string(ActionTypeURI):
+		ctx, err := r.newPdfActionURIFromDict(d)
+		if err != nil {
+			return nil, err
+		}
+		ctx.PdfAction = action
+		action.context = ctx
+		return action, nil
+	case string(ActionTypeSound):
+		ctx, err := r.newPdfActionSoundFromDict(d)
+		if err != nil {
+			return nil, err
+		}
+		ctx.PdfAction = action
+		action.context = ctx
+		return action, nil
+	case string(ActionTypeMovie):
+		ctx, err := r.newPdfActionMovieFromDict(d)
+		if err != nil {
+			return nil, err
+		}
+		ctx.PdfAction = action
+		action.context = ctx
+		return action, nil
+	case string(ActionTypeHide):
+		ctx, err := r.newPdfActionHideFromDict(d)
+		if err != nil {
+			return nil, err
+		}
+		ctx.PdfAction = action
+		action.context = ctx
+		return action, nil
+	case string(ActionTypeNamed):
+		ctx, err := r.newPdfActionNamedFromDict(d)
+		if err != nil {
+			return nil, err
+		}
+		ctx.PdfAction = action
+		action.context = ctx
+		return action, nil
+	case string(ActionTypeSubmitForm):
+		ctx, err := r.newPdfActionSubmitFormFromDict(d)
+		if err != nil {
+			return nil, err
+		}
+		ctx.PdfAction = action
+		action.context = ctx
+		return action, nil
+	case string(ActionTypeResetForm):
+		ctx, err := r.newPdfActionResetFormFromDict(d)
+		if err != nil {
+			return nil, err
+		}
+		ctx.PdfAction = action
+		action.context = ctx
+		return action, nil
+	case string(ActionTypeImportData):
+		ctx, err := r.newPdfActionImportDataFromDict(d)
+		if err != nil {
+			return nil, err
+		}
+		ctx.PdfAction = action
+		action.context = ctx
+		return action, nil
+	case string(ActionTypeSetOCGState):
+		ctx, err := r.newPdfActionSetOCGStateFromDict(d)
+		if err != nil {
+			return nil, err
+		}
+		ctx.PdfAction = action
+		action.context = ctx
+		return action, nil
+	case string(ActionTypeRendition):
+		ctx, err := r.newPdfActionRenditionFromDict(d)
+		if err != nil {
+			return nil, err
+		}
+		ctx.PdfAction = action
+		action.context = ctx
+		return action, nil
+	case string(ActionTypeTrans):
+		ctx, err := r.newPdfActionTransFromDict(d)
+		if err != nil {
+			return nil, err
+		}
+		ctx.PdfAction = action
+		action.context = ctx
+		return action, nil
+	case string(ActionTypeGoTo3DView):
+		ctx, err := r.newPdfActionGoTo3DViewFromDict(d)
+		if err != nil {
+			return nil, err
+		}
+		ctx.PdfAction = action
+		action.context = ctx
+		return action, nil
+	case string(ActionTypeJavaScript):
+		ctx, err := r.newPdfActionJavaScriptFromDict(d)
+		if err != nil {
+			return nil, err
+		}
+		ctx.PdfAction = action
+		action.context = ctx
+		return action, nil
+
+	}
+
+	common.Log.Debug("ERROR: Ignoring unknown action: %s", *actionType)
+	return nil, nil
+}
+
+func (r *PdfReader) newPdfActionGotoFromDict(d *core.PdfObjectDictionary) (*PdfActionGoTo, error) {
+	action := PdfActionGoTo{}
```

Comment:
Could just return here:

```go
return &PdfActionGoTo{
        D: d.Get("D"),
}, nil
```
Could do the same for the rest of the `newPdfActionXYZFromDict` methods as well.

Reply:
I followed the style of annotations. I will adapt it your proposed compact format